app/metadata.js

```diff
@@ -42,21 +42,27 @@ function getDocumentMetadata(url, window) {
 
 
 function getUrlMetadata(url) {
-  return fetch(url)
-    .then((res) => {
-      if (res.status >= 200 && res.status < 300) {
-        return res;
-      } else {
-        const error = new Error(res.statusText);
-        error.res = res;
-        throw error;
-      }
-    })
-    .then((res) => res.text())
-    .then((body) => getDocumentMetadata(url, domino.createWindow(body)))
-    .catch((err) => {
-      return {};
-    });
+  return new Promise((resolve) => {
+    const result = {url};
+    fetch(url)
+      .then((res) => {
+        if (res.status >= 200 && res.status < 300) {
```

Comment:
Fetch will follow redirects, so we expect to only receive 2xx in the case of a success.